rpc/blockchain.go

```diff
@@ -520,18 +424,41 @@ func (s *PublicBlockchainService) GetSignedBlocks(
 	ctx context.Context, address string,
 ) (interface{}, error) {
 	// Fetch the number of signed blocks within default period
-	totalSigned := uint64(0)
-	lastBlock := uint64(0)
-	blockHeight := s.hmy.CurrentBlock().Number().Uint64()
-	instance := shard.Schedule.InstanceForEpoch(s.hmy.CurrentBlock().Epoch())
-	if blockHeight >= instance.BlocksPerEpoch() {
-		lastBlock = blockHeight - instance.BlocksPerEpoch() + 1
-	}
-	for i := lastBlock; i <= blockHeight; i++ {
-		signed, err := s.IsBlockSigner(ctx, BlockNumber(i), address)
-		if err == nil && signed {
-			totalSigned++
+	curEpoch := s.hmy.CurrentBlock().Epoch()
+	var totalSigned uint64
+	if !s.hmy.ChainConfig().IsStaking(curEpoch) {
+		// calculate signed before staking epoch
+		totalSigned := uint64(0)
+		lastBlock := uint64(0)
+		blockHeight := s.hmy.CurrentBlock().Number().Uint64()
+		instance := shard.Schedule.InstanceForEpoch(s.hmy.CurrentBlock().Epoch())
+		if blockHeight >= instance.BlocksPerEpoch() {
+			lastBlock = blockHeight - instance.BlocksPerEpoch() + 1
+		}
+		for i := lastBlock; i <= blockHeight; i++ {
+			signed, err := s.IsBlockSigner(ctx, BlockNumber(i), address)
+			if err == nil && signed {
+				totalSigned++
+			}
+		}
+	} else {
+		ethAddr, err := internal_common.Bech32ToAddress(address)
```

Comment:
Yes. Previously, it will loop through the previous blocks in the same epoch to get signed rate (as in the `if` statement). This is actually a waste in performance compared with this method, which is to compare with last epoch snapshot.